src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -493,29 +467,18 @@ private boolean checkAccount(final AccountManager am, String username, String un
                 Account[] accountList = mAcctManager
                         .getAccountsByType(AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE);
 
-                // Authenticator installed from Company portal
-                // This supports only one account
+                // For new broker with PRT support, both company portal and
+                // azure authenticator will be able to support multi-user.
                 if (authenticator.packageName
-                        .equalsIgnoreCase(AuthenticationConstants.Broker.PACKAGE_NAME)) {
-                    // Adal should not connect if given username does not match
-                    if (accountList != null && accountList.length > 0) {
-                        return verifyAccount(accountList, username, uniqueId);
-                    }
-                    
-                    return false;                    
-
-                    // Check azure authenticator and allow calls for test
-                    // versions
-                } else if (authenticator.packageName
                         .equalsIgnoreCase(AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME)
                         || authenticator.packageName
-                                .equalsIgnoreCase(AuthenticationSettings.INSTANCE
-                                        .getBrokerPackageName())) {
-
+                                .equalsIgnoreCase(AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME)
+                        || authenticator.packageName
+                                .equalsIgnoreCase(AuthenticationSettings.INSTANCE.getBrokerPackageName())) {
```

Comment:
I didn't change the logic around here. We always has this check for testing purpose I think.